packages/shorebird_cli/lib/src/command_pipeline/pipeline_context.dart

```diff
@@ -0,0 +1,22 @@
+class PipelineContext {
+  PipelineContext() : _data = {};
+
+  PipelineContext._(Map<String, Object> data) : _data = data;
```

Comment:
Originally I thought on this being a class with defined attributes instead of a map.

That would mean that if a step needs to include a new attribute, they would need to change this class. But I assumed that this would not change that much and that would make this more compile safer?

Either way I also like the flexibility of a map brings, so no strong opinions

Reply:
Can you give examples of what you expected the properties to be?

Reply:
Something like:

```dart
class PipelineContext {
  final Directory projectRoot;
  final String? releaseVersion;
  final bool bundlePatchSignKey;
}
```

Idk if all those attributes makes sense, but were the ones that came to mind while trying to think in an example haha

Reply:
Gotcha. I think I'm going to start with a map to mimic the behavior of dart frog and provider, but we'll see if I end up fighting that too much.